src/main/scala/com/databricks/sparkdl/ImageUtils.scala

```diff
@@ -35,81 +47,88 @@ private[sparkdl] object ImageUtils {
     val height = ImageSchema.getHeight(rowImage)
     val width = ImageSchema.getWidth(rowImage)
     val channels = ImageSchema.getNChannels(rowImage)
+    val mode = ImageSchema.getMode(rowImage)
     val imageData = ImageSchema.getData(rowImage)
     require(
       imageData.length == height * width * channels,
       s"""| Only one byte per channel is currently supported, got ${imageData.length} bytes for
           | image of size ($height, $width, $channels).
        """.stripMargin
     )
-    val image = new BufferedImage(width, height, BufferedImage.TYPE_3BYTE_BGR)
 
-    var offset, h = 0
-    var r, g, b: Byte = 0
-    while (h < height) {
-      var w = 0
-      while (w < width) {
-        channels match {
-          case 1 =>
-            b = imageData(offset)
-            g = b
-            r = b
-          case 3 =>
-            b = imageData(offset)
-            g = imageData(offset + 1)
-            r = imageData(offset + 2)
-          case _ =>
-            require(false, s"`Channels` must be 1 or 3, got $channels.")
-        }
+    val imageType = openCVModeToImageType.getOrElse(mode,
+      throw new UnsupportedOperationException("Cannot convert row image with  " +
```

Comment:
@tomasatdatabricks As of right now, this error can only occur if users try to pass an image row with an unsupported OpenCV mode to Scala's DeepImageFeaturizer. I wonder if the current error message is vague/unhelpful to the user in this case...maybe the image validation should be pushed into DeepImageFeaturizer?

Reply:
Yes I think it should definitely be here as well. I believe you should always try to throw informative exception.   Deep ImageFeaturizer is just one of the possible users of this function. Also, it does not have to have the same restrictions.